### PR TITLE
feat: Product page shows list of categories associated with it

### PR DESCRIPTION
**Pull request by riqwan**
**What:**

Categories that the product is associated with is displayed in the product page as a detail.

Why:

Product categories feature UI completeness.

How:

- Creates a molecule for delimited list
- Reuse molecule in sales channel list in products page
- Product's Detail component now accepts a list that will delimit the list

**Visuals:**

Filled State:
Screenshot 2023-02-08 at 11 35 54

Empty State:
Screenshot 2023-02-08 at 11 41 49

RESOLVES CORE-1090

### CODE REVIEW

File: src/domain/products/edit/sections/general/index.tsx [hunk 1]
```diff
@@ -141,20 +158,6 @@ const ProductTags = ({ product }: Props) => {
   )
 }
 
-type SalesChannelBadgeProps = {
-  channel: SalesChannel
-}
-
-const SalesChannelBadge: React.FC<SalesChannelBadgeProps> = ({ channel }) => {
```
Comment (riqwan) [inline src/domain/products/edit/sections/general/index.tsx:148]: this doesn't seem to be used anywhere.

File: src/components/molecules/delimited-list/index.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        {list.slice(delimit).map((listItem) => (
+          <span key={listItem.id}>{listItem[property]}</span>
+        ))}
+      </div>
+    )
+  }
+
+  return (
+    <span className="inter-small-regular">
+      {list
```
Comment (adrien2p) [inline src/components/molecules/delimited-list/index.tsx:31]: suggetion: assign the sliced content to a const and create a boolean const such as `hasMore` with the constraint `list.length > delimit`

File: src/components/molecules/delimited-list/index.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  property = "name",
+}) => {
+  if (!list.length) {
+    return <></>
+  }
+
+  const ToolTipContent = () => {
+    return (
+      <div className="flex flex-col">
+        {list.slice(delimit).map((listItem) => (
```
Comment (adrien2p) [inline src/components/molecules/delimited-list/index.tsx:22]: suggestion: assign to a const and then use. Also, one question, does slicing here impact the component that use this mol?

File: src/domain/products/edit/sections/general/index.tsx [hunk 4]
```diff
@@ -99,22 +99,39 @@ type DetailProps = {
 }
 
 const Detail = ({ title, value }: DetailProps) => {
+  const DetailValue = () => {
```
Comment (adrien2p) [inline src/domain/products/edit/sections/general/index.tsx:102]: suggestion:

```ts

if (Array.isArray(value) && value.length) {
  return <DelimitedList list={value} delimit={2} />
} else {
  return <p><p/>
}

return <p>{value ? value : "–"}</p>
```

File: src/domain/products/edit/sections/general/index.tsx [hunk 5]
```diff
@@ -99,22 +99,39 @@ type DetailProps = {
...
+  const { isFeatureEnabled } = useFeatureFlag()
+
   return (
     <div className="flex flex-col mt-8 gap-y-3">
       <h2 className="inter-base-semibold">Details</h2>
       <Detail title="Subtitle" value={product.subtitle} />
       <Detail title="Handle" value={product.handle} />
       <Detail title="Type" value={product.type?.value} />
       <Detail title="Collection" value={product.collection?.title} />
+      {isFeatureEnabled("product_categories") && (
```
Comment (adrien2p) [inline src/domain/products/edit/sections/general/index.tsx:132]: suggestion: should that `"product_categories"` be exported from elsewhere as a const and use here?

**Review by adrien2p — CHANGES_REQUESTED** [review]

**Review by fPolic — APPROVED** [review]
LGTM!

File: src/components/molecules/delimited-list/index.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import React from "react"
+import Tooltip from "../../atoms/tooltip"
+
+type DelimitedListProps = {
```
Comment (adrien2p) [inline src/components/molecules/delimited-list/index.tsx:4]: suggestion: 
```ts
type DelimitedListProps<T = unknown> = {
  list: Record<string, T>[]
  delimit?: number
  property?: string
}
```

File: src/domain/products/edit/sections/general/index.tsx [hunk 7]
```diff
@@ -99,22 +99,39 @@ type DetailProps = {
 }
 
 const Detail = ({ title, value }: DetailProps) => {
```
Comment (pKorsholm) [inline src/domain/products/edit/sections/general/index.tsx:101]: Todo: Update type of `DetailProps.value` to include a list of items

File: src/domain/products/edit/sections/general/index.tsx (hunk 4 above)
Comment (pKorsholm) [reply src/domain/products/edit/sections/general/index.tsx:102]: remember the `-` on line 4 😄

**Review by pKorsholm — CHANGES_REQUESTED** [review]
It looks great! I have a single minor todo in addition to the suggestions already made. 😄

File: src/components/molecules/delimited-list/index.tsx (hunk 3 above)
Comment (riqwan) [reply src/components/molecules/delimited-list/index.tsx:22]: I think as long as the list changes, the slice will follow effect.

File: src/domain/products/edit/sections/general/index.tsx (hunk 4 above)
Comment (adrien2p) [reply src/domain/products/edit/sections/general/index.tsx:102]: Oho my bad i did made a mistake let me change it 😂

File: src/components/molecules/delimited-list/index.tsx (hunk 3 above)
Comment (adrien2p) [reply src/components/molecules/delimited-list/index.tsx:22]: What i mean is will the slice effect the list of the parent component

File: src/domain/products/edit/sections/general/index.tsx (hunk 4 above)
Comment (adrien2p) [reply src/domain/products/edit/sections/general/index.tsx:102]: Done @riqwan let me
Know what you think, it is nit so i let you choose

Comment (adrien2p) [reply src/domain/products/edit/sections/general/index.tsx:102]: It avoids sub if statements, makes more readable and easy to evolve

File: src/components/molecules/delimited-list/index.tsx (hunk 3 above)
Comment (riqwan) [reply src/components/molecules/delimited-list/index.tsx:24]: No, it doesn't. Especially not with the latest commit.

**Review by adrien2p — APPROVED** [review]
LGTM 🎉 just the pipeline now

File: src/domain/products/edit/sections/general/index.tsx [hunk 8]
```diff
@@ -95,26 +95,46 @@ const GeneralSection = ({ product }: Props) => {
...
+
+    if (value.length) {
+      return <DelimitedList list={value} delimit={2} />
+    }
+
+    return <p>–</p>
+  }
+
   return (
     <div className="flex items-center justify-between inter-base-regular text-grey-50">
```
Comment (pKorsholm) [inline src/domain/products/edit/sections/general/index.tsx:115]: Suggestion: ESLint should now sort classes (you might need to restart the linting server)

**Review by pKorsholm — APPROVED** [review]
LGTM with comments

It looks good! there's one type error otherwise just a couple of nits 😄

**Pull request merged**